java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java

```diff
@@ -23,38 +28,57 @@
-    @Override
-    public ResponseEntity<SearchMenuResult> searchMenu(final String searchTerm,
-                                                       final UUID restaurantId,
-                                                       final Integer pageSize,
-                                                       final Integer pageNumber) {
-        List<Menu> menuList;
-
-        if (isNotEmpty(searchTerm) && nonNull(restaurantId)) {
-            menuList = this.menuService.findAllByRestaurantIdAndNameContaining(
-                    restaurantId, searchTerm, pageSize, pageNumber);
-        } else if (isNotEmpty(searchTerm)) {
-            menuList = this.menuService.findAllByNameContaining(searchTerm, pageSize, pageNumber);
-        } else if (nonNull(restaurantId)) {
-            menuList = this.menuService.findAllByRestaurantId(restaurantId, pageSize, pageNumber);
-        } else {
-            menuList = this.menuService.findAll(pageNumber, pageSize);
-        }
-
-        return ResponseEntity.ok(new SearchMenuResult(pageSize, pageNumber,
-                menuList.stream().map(SearchMenuResultItem::new).collect(Collectors.toList())));
-    }
+  public MenuControllerImpl(MenuService menuService, ObjectMapper mapper) {
+    this.menuService = menuService;
+    this.mapper = mapper;
+  }
+
+  @Override
+  public ResponseEntity<SearchMenuResult> searchMenu(final String searchTerm,
+                                                     final UUID restaurantId,
+                                                     final Integer pageSize,
+                                                     final Integer pageNumber) {
+    List<Menu> menuList;
 
-    @Override
-    public ResponseEntity<Menu> getMenu(UUID id) {
-        return ResponseEntity.of(this.menuService.findById(id));
+    if (isNotEmpty(searchTerm) && nonNull(restaurantId)) {
+      menuList = this.menuService.findAllByRestaurantIdAndNameContaining(
+              restaurantId, searchTerm, pageSize, pageNumber);
+    } else if (isNotEmpty(searchTerm)) {
+      menuList =
+              this.menuService.findAllByNameContaining(searchTerm, pageSize, pageNumber);
+    } else if (nonNull(restaurantId)) {
+      menuList =
+              this.menuService.findAllByRestaurantId(restaurantId, pageSize, pageNumber);
+    } else {
+      menuList = this.menuService.findAll(pageNumber, pageSize);
     }
+
+    return ResponseEntity.ok(new SearchMenuResult(pageSize, pageNumber,
+            menuList.stream().map(SearchMenuResultItem::new).collect(Collectors.toList())));
+  }
+
+  @Override
+  public ResponseEntity<Menu> getMenu(UUID id) {
+    return ResponseEntity.of(this.menuService.findById(id));
+  }
+
+  /**
+   * @param menuCreateRequestDto menu to be created
+   * @return MenuCreatedDto
+   */
+  @Override
+  public ResponseEntity createMenu(
+          MenuCreateRequestDto menuCreateRequestDto) {
+    MenuCreatedDto menuCreatedDto =
+            this.menuService.saveMenu(menuCreateRequestDto);
```

Comment:
Given the DTO is for the api, the service layer shouldn't be aware of it. Only the Menu class.